sabnzbd/downloader.py

```diff
@@ -401,8 +402,12 @@ def decode(self, article, lines, raw_data):
         # See if there's space left in cache, pause otherwise
         # But do allow some articles to enter queue, in case of full cache
         qsize = self.decoder_queue.qsize()
-        if (not ArticleCache.do.reserve_space(lines) and qsize > MAX_DECODE_QUEUE) or (qsize > LIMIT_DECODE_QUEUE):
+
+        if (qsize > LIMIT_DECODE_QUEUE) or (lines and qsize > MAX_DECODE_QUEUE and not ArticleCache.do.reserve_space(lines)):
             sabnzbd.downloader.Downloader.do.delay()
+            return 0
+
+        return LIMIT_DECODE_QUEUE - qsize
```

Comment:
decode is rewritten to let the caller know if the queue is filled to fast, particularly related to articles that are out of retention. The if statement is rearranged to avoid calling ArticleCache.do.reserve_space if it isn't necessary.